packages/botonic-core/index.d.ts

```diff
@@ -105,20 +118,29 @@ export declare function getAvailableAgents(
   session: Session
 ): Promise<{ agents: HubtypeAgentsInfo[] }>
 
+export interface BotRequest {
+  input: Input
+  session: Session
+  lastRoutePath: string
+}
+
 /** The response of the bot for the triggered actions, which can be
  * the one matched by the routes, the default action and the retry actions.
- * See the @botonic/react's index.d.ts for this implementations particular type.
+ * See Response at @botonic/react's index.d.ts for the React type
  * */
-export type Response = any
+export interface BotResponse extends BotRequest {
+  response: any
+}
+
+export type PluginPreRequest = BotRequest
+export type PluginPostRequest = BotRequest & {
```

Comment:
Use maybe BotResponse instead of BotRequest & {response: any}?

Reply:
yes, I'll do it in another PR